strawberry/channels/handlers/graphql_ws_handler.py

```diff
@@ -0,0 +1,54 @@
+from contextlib import suppress
+from typing import Any, Optional
+
+from strawberry.channels.handlers.base import ChannelsWSConsumer
+from strawberry.schema import BaseSchema
+from strawberry.subscriptions import GRAPHQL_WS_PROTOCOL
+from strawberry.subscriptions.protocols.graphql_ws.handlers import BaseGraphQLWSHandler
+from strawberry.subscriptions.protocols.graphql_ws.types import OperationMessage
+
+
+class GraphQLWSHandler(BaseGraphQLWSHandler):
+    def __init__(
+        self,
+        schema: BaseSchema,
+        debug: bool,
+        keep_alive: bool,
+        keep_alive_interval: float,
+        get_context,
+        get_root_value,
+        ws: ChannelsWSConsumer,
+    ):
+        super().__init__(schema, debug, keep_alive, keep_alive_interval)
+        self._get_context = get_context
+        self._get_root_value = get_root_value
+        self._ws = ws
+
+    async def get_context(self) -> Any:
+        return await self._get_context(request=self._ws)
+
+    async def get_root_value(self) -> Any:
+        return await self._get_root_value(request=self._ws)
+
+    async def send_json(self, data: OperationMessage) -> None:
+        await self._ws.send_json(data)
+
+    async def close(self, code: int = 1000, reason: Optional[str] = None) -> None:
+        # Close messages are not part of the ASGI ref yet
+        await self._ws.close(code=code)
+
+    async def handle_request(self) -> Any:
+        await self._ws.accept(subprotocol=GRAPHQL_WS_PROTOCOL)
+
+    async def handle_disconnect(self, code):
+        if self.keep_alive_task:
+            self.keep_alive_task.cancel()
+            with suppress(BaseException):
+                await self.keep_alive_task
+
+        for operation_id in list(self.subscriptions.keys()):
+            await self.cleanup_operation(operation_id)
+
+    async def handle_invalid_message(self, error_message: str) -> None:
```

Comment:
Do we need this method? (it's not part of the `BaseGraphQLWSHandler`)

Reply:
The way the [consumer is implemented](https://github.com/strawberry-graphql/strawberry/pull/1407/files#diff-b1a5f5f9278dcb583df371f041d6b836724977d0b9c0781251a0e9ff70e6336fR101), it calls the `handle_invalid_message` from the handler in a generic way, so this is here just to "do nothing" since `BaseGraphQLWSHandler` indeed doesn't implement this.

I can check there to see if the current handler is an instance of `BaseGraphQLWSHandler` instead if you think it is better :)

Reply:
Ah I see. Since it confused me for a second it might be worth adjusting it. Using `isinstance`, `hasattr` or adding an explanation to that method all sound good to me :)

Reply:
I added an explanation there! Please tell me if you prefer another route